src/library/trackcollection.cpp

```diff
@@ -510,4 +514,80 @@ StorageUnicodeFunctions::likeFunction(sqlite3_context *p,
                                     itrString, endString, E));
 }
 */
+//Private function
+
+bool TrackCollection::settingsPathStatus(){
+    QFileInfo fle(m_pConfig->getSettingsPath());
+    QFile::Permissions permit = fle.permissions();
+    permit = permit & MASK;
+    const short int NONE = QFile::NoOptions;
+    const short int EXE = QFile::ExeOwner;
```

Comment:
no, i checked that if the path is not executable then this don't work, it shows the error message unable to connect to the database, and executable folder means we can go in it so it is necessary